plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-orders-controller.php

```diff
@@ -205,6 +205,7 @@ protected function save_object( $request, $creating = false ) {
 			if ( $creating ) {
 				$object->set_created_via( 'rest-api' );
 				$object->set_prices_include_tax( 'yes' === get_option( 'woocommerce_prices_include_tax' ) );
+				$object->save();
```

Comment:
It'd be good to add a test case as well which reproduces the problem so that this fix shows the problem as fixed.